protocol/topic.go

```diff
@@ -157,7 +157,12 @@ func (topic *Topic) UnmarshalJSON(data []byte) error {
 	topic.Criterion = TopicCriterion(elements[index])
 	index++
 	if index < len(elements) {
-		topic.Action = TopicAction(elements[index])
+		if topic.Criterion == CriterionMessages {
+			i := strings.Index(v, elements[index])
+			topic.Action = TopicAction(v[i:])
+		} else {
+			topic.Action = TopicAction(elements[index])
+		}
 	} else {
 		topic.Action = ""
 	}
```

Comment:
If only one regexp. is used - the logic here can be simplified (i.e. omit manual string splitting and concatenating) and rely on the matching groups in the regexp.
Please, see - func (*Regexp) [FindAllStringSubmatch](https://cs.opensource.google/go/go/+/go1.17.7:src/regexp/regexp.go;l=1190)